### PR TITLE
update eks versions and remove duplicate printer fields

### PR DESCRIPTION
**Pull request by nabuskey**
### What does this PR do?

1. Removes commented out fields
2. Add newer EKS versions to the enum
3. Add fields to support aws-auth config map management

### CODE REVIEW

File: compositions/aws-provider/eks/definition.yaml [hunk 1]
```diff
@@ -43,11 +37,10 @@ spec:
                   description: EKS Input parameters
                   type: object
                   properties:
-
                     version:
                       description: Kubernetes Version
                       type: string
-                      enum: [ "1.21", "1.22", "1.23" ]
+                      enum: [ "1.21", "1.22", "1.23", "1.24", "1.25" ]
```
Comment (candonov) [inline compositions/aws-provider/eks/definition.yaml:43]: Can you remove 1.21 while you are it?

File: compositions/aws-provider/eks/eks-managed-node-group-subnet-labels.yaml [hunk 2]
```diff
@@ -75,13 +75,11 @@ spec:
...
+        - type: ToCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: status.clusterName
         - fromFieldPath: spec.writeConnectionSecretToRef.namespace
           toFieldPath: spec.writeConnectionSecretToRef.namespace
-#        - type: ToCompositeFieldPath
-#          fromFieldPath: status.atProvider.identity.oidc.issuer
-#          toFieldPath: status.eks.oidc
-#          policy:
-#            fromFieldPath: Optional
```
Comment (candonov) [inline compositions/aws-provider/eks/eks-managed-node-group-subnet-labels.yaml:84]: Do you know why these fields we commented out? And why we are removing them vs uncommenting?

Comment (nabuskey) [reply compositions/aws-provider/eks/eks-managed-node-group-subnet-labels.yaml:84]: Not too sure tbh. I think this was left over from our initial commit. As far as I am aware, it hasn't been used.

File: compositions/aws-provider/eks/definition.yaml (hunk 1 above)
Comment (nabuskey) [reply compositions/aws-provider/eks/definition.yaml]: Removed 1.21

**Review by candonov — APPROVED** [review]

**Pull request merged**